### PR TITLE
feat(ms-teams): Add document handling for sending message to channel

### PR DESCRIPTION
**Pull request by ztefanie**
## Description
Added document support when sending messages to channels

See example bpmn [here](https://modeler.ultrawombat.com/diagrams/fae5f90c-edc9-4c9e-81af-d3120555f1b8--teams?v=579,282,1)
Get your token [here](https://developer.microsoft.com/en-us/graph/graph-explorer) under "Access Token" tab

Documented here: https://github.com/camunda/camunda-docs/pull/5333

## Related issues

partially solves https://github.com/camunda/team-connectors/issues/1002

In a next PR the option to send documents in private messages is added, this is currently not possible, because of access restrictions in our MS Teams setup. We are hoping to get access to a Sandbox environment where we can test this. When sending documents in messages, the documents are uploaded to the SharePoint. When sending documents to a channel, they are uploaded to the teams sharepoint. When sending documents to a private chat, they are uploaded to the private sharepoint. I currently assume, that bots / apps do not have own sharepoints and therefore need to write the documents they want to send, to the receivers SharePoint. As I cannot test this with the current MS teams options / permissions, this is postponed. 

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

**Review by chillleader — DISMISSED** [review]
Awesome!

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            .filter(Objects::nonNull)
+            .map(
+                document -> {
+                  try {
+                    return uploadDocument(
+                        graphClient, model.groupId(), document, document.metadata().getFileName());
+                  } catch (IOException
+                      | InvocationTargetException
+                      | IllegalAccessException
+                      | NoSuchMethodException e) {
```
Comment (chillleader) [inline connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java:51]: nit: wdyt if we handle these exceptions inside the `uploadDocument` method, so that we have all exceptions caught on the same level? The stream will also look nicer here

File: connectors/microsoft-teams/src/test/java/io/camunda/connector/model/request/channel/SendMessageToChannelTest.java [hunk 2]
```diff
@@ -52,7 +52,11 @@ public void init() {
 
     sendMessageToChannel =
         new SendMessageToChannel(
-            ActualValue.Channel.GROUP_ID, ActualValue.Channel.CHANNEL_ID, "channel content", null);
+            ActualValue.Channel.GROUP_ID,
+            ActualValue.Channel.CHANNEL_ID,
+            "channel content",
+            null,
+            null);
```
Comment (johnBgood) [inline connectors/microsoft-teams/src/test/java/io/camunda/connector/model/request/channel/SendMessageToChannelTest.java:59]: can we have some tests with Documents?

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CancellationException;
+
+public class DocumentHandler {
+
+  SendMessageToChannel model;
+
+  public DocumentHandler(SendMessageToChannel model) {
```
Comment (johnBgood) [inline connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java:35]: I would add the `GraphServiceClient graphClient` in this constructor and would remove all the params from `handleDocuments`

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    this.model = model;
+  }
+
+  public void handleDocuments(GraphServiceClient graphClient, ChatMessage chatMessage) {
```
Comment (johnBgood) [inline connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java:39]: here as I said I would remove all params, and I'd return `List<ChatMessageAttachment>`.

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+import io.camunda.document.Document;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CancellationException;
+
+public class DocumentHandler {
```
Comment (johnBgood) [inline connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java:31]: Should we have some basic UTs for this class?

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/SendMessageToChannelOperation.java [hunk 6]
```diff
@@ -25,6 +25,10 @@ public Object invoke(final GraphServiceClient graphClient) {
             .map(type -> BodyType.forValue(type.toLowerCase(Locale.ROOT)))
             .orElse(BodyType.Text));
     body.setContent(model.content());
+    if (model.documents() != null) {
+      DocumentHandler documentHandler = new DocumentHandler(model);
+      documentHandler.handleDocuments(graphClient, chatMessage);
```
Comment (johnBgood) [inline connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/SendMessageToChannelOperation.java:30]: Following my other comments, this would return a list of attachments.
We can then just `chatMessage.setAttachments(..)` here. It'd be consistent with how we deal with the body etc.

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java (hunk 1 above)
Comment (ztefanie) [reply connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java:51]: Good point thanks. I moved it.

**Review by johnBgood — APPROVED** [review]
Approving ahead of the UTs implementation as I guess I won't need to approve again for tests. 
Great work 💯

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Test required in (you can check both):
  - [ ] SaaS
  - [ ] SM
- [ ] Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*

File: connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java (hunk 5 above)
Comment (ztefanie) [reply connectors/microsoft-teams/src/main/java/io/camunda/connector/operation/channel/DocumentHandler.java:30]: For transparency: Jonathan and I decided UT do not make sense here, as all methods only call MS team functions and do not really have own logic.

File: connectors/microsoft-teams/src/test/java/io/camunda/connector/model/request/channel/SendMessageToChannelTest.java (hunk 2 above)
Comment (ztefanie) [reply connectors/microsoft-teams/src/test/java/io/camunda/connector/model/request/channel/SendMessageToChannelTest.java:59]: For transparency: Jonathan and I decided UT do not make sense here, as all methods only call MS team functions and do not really have own logic.

**Review by Szik — APPROVED** [review]
